var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -207,7 +207,12 @@ def configure(self, spec, prefix):
         # variable to set AMD_ARCH configure option.
         # Spack user can not directly use AMD_ARCH for this purpose but should
         # use target variable to set appropriate -march option in AMD_ARCH.
-        options.append(f"AMD_ARCH={optimization_flags(self.compiler, spec.target)}")
+        options.append(f"AMD_ARCH={optimization_flags(self.compiler, spec.target).split('=')[-1]}")
```

Comment:
Ugh, I though this option was meant to inject flags into compilation. Instead I see that it just accepts a fixed set of values:

https://github.com/amd/amd-fftw/blob/cfcb67c1f225c807d67dd99ad4f566d0a3eaf0f2/CMakeLists.txt#L13-L33

Should we add:
```
requires("target=zen:")
```
to the recipe? That would avoid the current state, where the recipe would just fail to build on Intel chips, or older AMD chips, right? Instead it would fail in concretization.

Reply:
With the CFLAGS, the code path in the amdfftw build system will bypass the logic around AMD_ARCH.

Reply:
Reading the `CMakeLists.txt` above, wouldn't this result in:
```
SET(AMD_ARCH "znver1" CACHE STRING "select AMD zen version for Clang toolchain")
```
being executed, and thus injecting:
```
-march=znver1 -mprefer-vector-width=XXX
```
during the build? In this case wouldn't this break on Intel chips? Was:
```
requires("target=zen:")
```
having some issue?

Reply:
We are still using autotools to build I believe, since the FFTWBase package inherits from autotools and we don't attempt to change that: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/fftw/package.py#L12

In the autotools configure script injection of custom CFLAGS will short-circuit all of the platform detection logic.  In which case the Spack-provided arch flags will make it through and be used.

We could constrain to zen* only, but we are trying to allow build+run across all x86_64 platforms to allow as wide an audience as possible to try our packages.

Reply:
Thanks for the explanation. LGTM then.